browser/brave_rewards/BUILD.gn

```diff
@@ -0,0 +1,61 @@
+import("//brave/components/brave_ads/browser/buildflags/buildflags.gni")
+import("//brave/components/brave_rewards/browser/buildflags/buildflags.gni")
+import("//extensions/buildflags/buildflags.gni")
+
+source_set("brave_rewards") {
+  sources = [
+    "rewards_service_factory.cc",
+    "rewards_service_factory.h",
+    "rewards_tab_helper.cc",
+    "rewards_tab_helper.h",
+  ]
+
+  deps = [
+    "//base",
+    "//brave/common",
+    "//brave/components/brave_rewards/browser",
+    "//brave/components/brave_rewards/common",
+    "//components/keyed_service/content",
+    "//components/prefs",
+    "//content/public/browser",
+    "//extensions/buildflags",
+  ]
+
+  if (enable_extensions) {
+    deps += [
+      "//extensions/browser",
+    ]
+  }
+
+  if (!is_android) {
+    sources += [
+      "tip_dialog.cc",
+      "tip_dialog.h",
+    ]
+
+    deps += [
+      "//components/guest_view/browser",
+      "//components/web_modal",
+      "//ui/web_dialogs",
+    ]
+  } else {
+    if (brave_rewards_enabled) {
```

Comment:
why do we need this check?

Reply:
not sure I understand, why wouldn't this be in a `brave_rewards_enabled` guard?

Reply:
actually the issue I see is that the java files aren't properly guarded for the buildflags or internally in the java code cc @NejcZdovc @SergeyZhukovsky

Reply:
I'm pretty sure that with or without this guard the build would break if it brave rewards was disabled

Reply:
i mean why do we add the `brave_rewards` dependency to `browser/BUILD.gn` if rewards are disabled? We don't need this check if the whole file is not used when rewards are disabled.

Reply:
The factory still exists if it is disabled, it just returns null. I think too often we have code that requires external ifdef guards that could more easily be handled internally. That makes things easier for people using the code because they don't have to worry about what needs guards and what doesn't.

Reply:
you end up with buildflags and guards spread all over the code and usually with missing deps for the buildflags, etc..